BrainPortal/lib/boutiques_create_fake_parent_dir.rb

```diff
@@ -0,0 +1,139 @@
+#
+
+# Some tools expect that a file/directory input is
+# placed input a parent directory.
+#
+# For example we have the following Boutiques command-line:
+#
+#     apptool [PRECOMPUTED_INPUT]
+#
+# with the following input in Boutiques:
+#
+#     {
+#       "description": "A directory that contain the precomputed data process by...",
+#       "id": "precomputed_input",
+#       "name": "Precomputed input",
+#       "optional": false,
+#       "type": "File",
+#       "value-key": "[PRECOMPUTED_INPUT]"
+#     }
+#
+# and in the `cbrain:integrator_modules` section:
+#
+#     "BoutiquesCreateFakeParentDir": ["precomputed_input"],
+#
+# In CBRAIN the user will select a userfile for example `sub-n`,
+# the final command line will become:
+#
+#     apptool fake_parent_dir/sub-n
+#
+module BoutiquesCreateFakeParentDir
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  ############################################
+  # Portal Side Modifications
+  ############################################
+
+  # Adjust the description for the input with the mention of
+  # the fake parent directory
+  def descriptor_for_form #:nodoc:
+    descriptor                = super.dup
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesCreateFakeParentDir')
+
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      fake_parent_dirname = fake_parent_dirname[0] if fake_parent_dirname.is_a?(Array)
+      if !Userfile.is_legal_filename?(fake_parent_dirname)
+        fake_parent_dirname = "fake_dir_uniq_id"
+      end
+
+      # Adjust the description
+      input              = descriptor.input_by_id(inputid)
+      input_description  = input.description || "";
+      add_to_description = "This input will be copied in a parent folder `#{fake_parent_dirname}`.The parent folder will be used in the command line"
+
+      input.description  = [input_description, add_to_description].compact.join("\n")
+    end
+
+    descriptor
+  end
+
+  def after_form
+    after_form_msg = super
+    return after_form_msg if self.errors.any?
+
+    descriptor = descriptor_for_form
+    parent_dirname_by_inputid = descriptor.custom_module_info('BoutiquesCreateFakeParentDir')
+
+    params["BoutiquesCreateFakeParentDir"] = {}
+    parent_dirname_by_inputid.each do |inputid,fake_parent_dirname|
+      params["BoutiquesCreateFakeParentDir"][inputid] = invoke_params[inputid]
+      invoke_params[inputid] = nil
+    end
+
+    after_form_msg
+  end
+
+  ############################################
+  # Bourreau (Cluster) Side Modifications
+  ############################################
+
+  # For input in the `BoutiquesCreateFakeParentDir` section,
+  # create a fake parent directory that will contains a symlink
+  # to the orginal selected Userfile.
+  def setup #:nodoc:
```

Comment:
I think the overall solution of setting aside the IDs of the input file to prevent the standard setup() to try to run make_available() on them is correct, but we're doing it the wrong way (and yes I know it was my suggestion). We can't set them aside in the after_form() method on the portal side. If we do that, the system won't work for task arrays (e.g. when a CbrainFileList is provided as an input for a File). 

The code can be adjusted in a much simpler way: everything is done in setup(), by insterting code BEFORE and AFTER the super() call:

```ruby
   def setup
      remove_ids_from_invoke_struct_here
      result = super
      reinstert_ids_here
      return result if ! result
      do_special_make_available_here
      return true or false # depending on maybe other checks
   end
```